test/Microsoft.AspNet.Mvc.FunctionalTests/LocalizationTest.cs

```diff
@@ -75,5 +78,103 @@ public static IEnumerable<object[]> LocalizationData
             // Assert
             Assert.Equal(expected, body.Trim(), ignoreLineEndingDifferences: true);
         }
+
+        public static IEnumerable<object[]> LocalizationResourceData
+        {
+            get
+            {
+                // Dnx does not support reading resources yet. Coreclr return null value while trying to read resources.
+                // https://github.com/aspnet/Mvc/issues/2747
+#if DNX451
+                var expected1 =
+@"Hello there!!
+Learn More
+Hi John      ! You are in 2015 year and today is Thursday";
+
+                yield return new[] {"en-GB", expected1 };
+
+                var expected2 =
+@"Bonjour!
+apprendre Encore Plus
+Salut John      ! Vous êtes en 2015 an aujourd'hui est Thursday";
+                yield return new[] { "fr", expected2 };
+#else
+                var expectedCoreClr =
+@"Hello there!!
+Learn More
+Hi";
+                yield return new[] {"en-GB", expectedCoreClr };
+                yield return new[] {"fr", expectedCoreClr };
+#endif
+
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(LocalizationResourceData))]
+        public async Task Localization_Resources_ReturnExpectedValues(string value, string expected)
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, _configureServices);
+            var client = server.CreateClient();
+            var cultureCookie = "c=" + value + "|uic=" + value;
+            client.DefaultRequestHeaders.Add(
+                "Cookie",
+                new CookieHeaderValue("ASPNET_CULTURE", cultureCookie).ToString());
+
+            if (!value.StartsWith("en"))
+            {
+                // Manually generating .resources file since we don't autogenerate .resources file yet. 
+                WriteResourceFile("HomeController." + value + ".resx");
+                WriteResourceFile("Views.Shared._LocalizationLayout.cshtml." + value + ".resx");
+            }
+            WriteResourceFile("Views.Home.Locpage.cshtml." + value + ".resx");
+
+            // Act
+            var body = await client.GetStringAsync("http://localhost/Home/Locpage");
+
+            // Assert
+            Assert.Equal(expected, body.Trim());
+        }
+
+        private void WriteResourceFile(string resxFileName)
+        {
+            var resxFilePath = Directory.GetParent(Directory.GetCurrentDirectory()) + "\\WebSites\\" + SiteName + 
+                "\\Resources\\" + resxFileName;
```

Comment:
Do not use platform-specific directory separators and don't bother with `Directory.GetParent()` since `resxFileName` is just a file name, not a more complex relative path. Instead use `Path.Combine("..", "WebSites", SiteName, "Resources")`.